packages/apollo-server-codegen/src/IR/Descriptions.ts

```diff
@@ -0,0 +1,13 @@
+import { Translatable, Translator } from "../Translators";
+
+export class Description implements Translatable {
```

Comment:
As a general comment for this PR, I think it would be incredibly valuable to add some documentation to these files signalling their intention. Take for example the interaction between `Translator` and `Description`, I'm not sure on initial read what that relationship gains. For future improvements and care of this addition I think we need more explanation as to the structure and relationship between the types.

Reply:
I added some docs and a architecture.md document, lmk what you think!